climate-monitoring-client/src/main/java/climatemonitoring/EditProfile.java

```diff
@@ -1,18 +1,125 @@
 package climatemonitoring;
 
+import climatemonitoring.core.ConnectionLostException;
+import climatemonitoring.core.DatabaseRequestException;
+import climatemonitoring.core.Operator;
 import climatemonitoring.core.ViewState;
+import climatemonitoring.core.headless.Console;
 
 class EditProfile extends ViewState {
 
 	@Override
 	public void onHeadlessRender(String args) {
 
-		throw new UnsupportedOperationException("Unimplemented method 'onHeadlessRender'");
+		Console.write("User ID > " + m_userID);
+		Console.write("SSID > " + m_ssid);
+		Console.write("Surname > " + m_surname);
+		Console.write("Name > " + m_name);
+		Console.write("Email > " + m_email);
+		Console.write("Password > " + m_password);
+		Console.write("Center ID > " + m_centerID);
+
+		String errorMsg = null;
+
+		do {
+
+			String newSSID = Console.read("New SSID > ");
+
+			if (newSSID != m_ssid) {
+
+				errorMsg = Check.ssidUnique(m_ssid);
+				m_ssid = newSSID;
+	
+				if (errorMsg != null)
+					Console.write(errorMsg);
+			}
+
+		} while (errorMsg != null);
+
+		do {
+
+			m_surname = Console.read("New Surname > ");
+			errorMsg = Check.surname(m_surname);
+
+			if (errorMsg != null)
+				Console.write(errorMsg);
+
+		} while (errorMsg != null);
+
+		do {
+
+			m_name = Console.read("New Name > ");
+			errorMsg = Check.name(m_name);
+
+			if (errorMsg != null)
+				Console.write(errorMsg);
+
+		} while (errorMsg != null);
+
+		do {
+
+			String newEmail = Console.read("New Email > ");
+
+			if (newEmail != m_email) {
+
+				errorMsg = Check.emailUnique(m_email);
+				m_email = newEmail;
```

Comment:
Just use the `Check.email()` method, because in a few days it will be the same as `Check.emailUnique()` + this last one will be removed

Reply:
Fixed